src/main/java/DrukmakoriSivatag/Plumber.java

```diff
@@ -146,42 +146,36 @@ public boolean addPump(){
      * már csak annyi történik, hogy a Plumber objektum pickedUpPipe attribútumát egyenlővé
      * teszi azzal a csővel, amit lecsatolt.
      *
-     * @param idx   A lecsatlakoztatni kívánt cső indexe.
-     *@return Sikeres volt-e a művelet.
-     *
+     * @param idx A lecsatlakoztatni kívánt cső indexe.
+     * @return Sikeres volt-e a művelet.
      */
     @Override
-    public boolean pickUpPipeEnd(int idx){
-        boolean success = false;
-        if(pickedUpPipe==null){
+    public boolean pickUpPipeEnd(int idx) {
+        if (pickedUpPipe == null) {
             setPickedUpPipe(element.pickUpPipeEnd(idx));
-            if(pickedUpPipe != null){
-                success = true;
-            }
+            return pickedUpPipe != null;
         }
-        return success;
+
+        return false;
     }
 
     /**
-     *
      * Egy teljes cső lecsatlakoztatásáért felelős. Az új pumpa felvételéhez hasonlóan itt is megvizsgálja,
      * hogy nincs-e lecsatlakoztatott végű cső már a játékosnál. Ha nincs nála, akkor
      * meghívja az element tagváltozó azonos nevű függvényét. Ezt csak a pumpa
      * implementálja, hiszen csak arról lehet csövet lecsatlakoztatni teljesen.
      *
-     * @param idx   A lecsatlakoztatni kívánt cső indexe.
+     * @param idx A lecsatlakoztatni kívánt cső indexe.
      * @return Sikeres volt-e a művelet.
      */
     @Override
-    public boolean pickUpPipe(int idx){
-        boolean success = false;
-        if(pickedUpPipe==null){
+    public boolean pickUpPipe(int idx) {
+        if (pickedUpPipe == null) {
```

Comment:
negate this condition and return if it is not met, so we remove one nesting from the code.